docs/DesignDocs/edl_security_and_portability.md

```diff
@@ -0,0 +1,109 @@
+EDL security and portability enhancements
+=========================================
+
+Motivation
+----------
+
+This document proposes EDL enhancements that will:
+
+- Improve portability to TrustZone
+- Mitigate security vulnerabilities
+
+This feature is needed by TrustZone to ensure that applications developed with
+EDL will work on TrustZone platforms. Currently the use of host shared memory
+is problematic for TrustZone, which utilizes a different shared memory model.
+SGX *implicitly* shares *all* host memory with the enclave, whereas, TrustZone
+*explicitly* shares *some* host memory with the enclave. Therefore we
+distinguish two models.
```

Comment:
Wording issue: There may be other models for other TEEs.   For example, some likely share "none".  (For example today Windows use of TrustZone is in the "none" category, and secure elements might be in that category too.)

Reply:
Good point. I'll update and add the "none" category.